cmd/syncer/main.go

```diff
@@ -216,6 +219,50 @@ func initSyncerComponents(ctx context.Context, clusterFlavor cnstypes.CnsCluster
 				log.Errorf("failed to initialize nodeManager. Error: %+v", err)
 				os.Exit(1)
 			}
+			if configInfo.Cfg.Global.ClusterDistribution == "" {
+				config, err := rest.InClusterConfig()
+				if err != nil {
+					log.Errorf("failed to get InClusterConfig: %v", err)
+					os.Exit(1)
+				}
+				clientset, err := kubernetes.NewForConfig(config)
+				if err != nil {
+					log.Errorf("failed to create kubernetes client with err: %v", err)
+					os.Exit(1)
+				}
+
+				// Get the version info for the Kubernetes API server
+				versionInfo, err := clientset.Discovery().ServerVersion()
+				if err != nil {
+					log.Errorf("failed to fetch versionInfo with err: %v", err)
+					os.Exit(1)
+				}
+
+				// Extract the version string from the version info
+				version := versionInfo.GitVersion
+				var ClusterDistNameToServerVersion = map[string]string{
+					"gke":       "Anthos",
+					"racher":    "Rancher",
```

Comment:
`racher` looks like a typo.

Reply:
it's the upstream code, I don't want to introduce differences here